File: backend/src/main/java/com/clb/charity/event/domain/Event.java
package com.clb.charity.event.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.jspecify.annotations.Nullable;

import java.time.Instant;
import java.time.LocalDate;

@Entity
@Table(name = "events")
@Getter
@Setter
@NoArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private @Nullable Long id;

    @Column(nullable = false)
    private String title;

    @Column(name = "title_en", length = 255)
    private @Nullable String titleEn;

    @Column(columnDefinition = "text")
    private @Nullable String description;

    @Column(name = "description_en", columnDefinition = "text")
    private @Nullable String descriptionEn;

    @Column(name = "event_start_date", nullable = false)
    private LocalDate eventStartDate;

    @Column(name = "event_end_date")
    private @Nullable LocalDate eventEndDate;

    @Column(length = 255)
    private @Nullable String location;

    /** Author's member id — referenced by id only, no JPA relation (module boundary). */
    @Column(name = "created_by")
    private @Nullable Long createdBy;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private @Nullable Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private @Nullable Instant updatedAt;
}

File: frontend/src/types/event.ts
// ---- Internal activities (non-fundraising events) ----

export interface Event {
  id: number
  title: string
  titleEn: string | null
  description: string | null
  descriptionEn: string | null
  eventStartDate: string
  eventEndDate: string | null
  location: string | null
  createdBy: number | null
  createdAt: string
  updatedAt: string
}

export interface CreateEventRequest {
  title: string
  titleEn: string | null
  description: string | null
  descriptionEn: string | null
  eventStartDate: string
  eventEndDate: string | null
  location: string | null
}

export type UpdateEventRequest = CreateEventRequest
